### PR TITLE
new Dashboard 

### PR DESCRIPTION
**Pull request by MohsenSaeidiDev**

### CODE REVIEW

File: src/traits/CoreTrait.php [hunk 1]
```diff
@@ -246,4 +287,88 @@ protected function firstMonthNumberOfSeason($monthNumber)
...
+
+    private function getPersianDateComponents($pdate) {
+        $dateArray = explode('/', $pdate->jdate('Y/m/d'));
+        return array_map([$pdate, 'persian_to_english'], $dateArray);
+    }
+
+    protected function getStartAndEndOfLastMonth(){
+
+        $pdate = Yii::$app->pdate;
+        list($jYear, $jMonth, $jDay) = $this->getPersianDateComponents($pdate);
```
Comment (Sadi01) [inline src/traits/CoreTrait.php]: list($jYear, $jMonth, $jDay) = $pdate->jdate('Y/m/d', time(), '', 'Asia/Tehran', 'en');

File: src/traits/CoreTrait.php [hunk 2]
```diff
@@ -246,4 +287,88 @@ protected function firstMonthNumberOfSeason($monthNumber)
         }
         return null;
     }
+
+    private function getPersianDateComponents($pdate) {
```
Comment (Sadi01) [inline src/traits/CoreTrait.php]: Remove

File: src/components/Pdate.php [hunk 3]
```diff
@@ -662,6 +662,14 @@ function jalali_to_gregorian($jy, $jm, $jd, $mod = '')
         }
         return ($mod === '') ? array($gy, $gm, $gd) : $gy . $mod . $gm . $mod . $gd;
     }
+
+    function persian_to_english($string) {
```
Comment (Sadi01) [inline src/components/Pdate.php]: Remove

File: src/traits/CoreTrait.php [hunk 4]
```diff
@@ -246,4 +287,88 @@ protected function firstMonthNumberOfSeason($monthNumber)
...
+        return [
+            'start' => $firstDayLastMonth,
+            'end' => $lastDayLastMonth
+        ];
+    }
+
+    protected function getStartAndEndOfLastYear()
+    {
+        $pdate = Yii::$app->pdate;
+        $jYear = $this->getPersianDateComponents($pdate)[0];
```
Comment (Sadi01) [inline src/traits/CoreTrait.php]: $jYear = $pdate->jdate('Y/m/d', time(), '', 'Asia/Tehran', 'en');

**Review by Sadi01 — CHANGES_REQUESTED** [review]

File: src/traits/CoreTrait.php (hunk 1 above)
Comment (MohsenSaeidiDev) [reply src/traits/CoreTrait.php]: Done!

File: src/traits/CoreTrait.php (hunk 2 above)
Comment (MohsenSaeidiDev) [reply src/traits/CoreTrait.php]: Removed!

File: src/traits/CoreTrait.php (hunk 4 above)
Comment (MohsenSaeidiDev) [reply src/traits/CoreTrait.php]: Replaced!

**Review by Sadi01 — APPROVED** [review]

**Pull request merged**